lib/rubocop/cop/rails/time_zone.rb

```diff
@@ -135,7 +135,7 @@ def check_time_node(klass, node)
         end
 
         def attach_timezone_specifier?(date)
-          date.respond_to?(:value) && TIMEZONE_SPECIFIER.match?(date.value.to_s)
+          date.respond_to?(:value) && (!date.value.to_s.valid_encoding? || TIMEZONE_SPECIFIER.match?(date.value.to_s))
```

Comment:
That's just my two cents:

```suggestion
          return false unless date.respond_to?(:value)

          !date.value.to_s.valid_encoding? || TIMEZONE_SPECIFIER.match?(date.value.to_s)
```